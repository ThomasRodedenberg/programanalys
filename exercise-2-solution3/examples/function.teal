fun f() : nonnull int = { return 0; }

fun g() = { return null; }

fun main() = {
    var x : nonnull int := f(); // NC 6 27, N 6 4 +
    var y : nonnull int := g(); // NC 7 27, N 7 4 ?
}
