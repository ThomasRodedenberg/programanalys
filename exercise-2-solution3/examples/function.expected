maybe: N 1 33 +
N 6 4 +
N 7 4 ?
